src/fsql/column_parser.py

```diff
@@ -84,13 +84,25 @@ def process_single_partition(partition_desc: str):
 
 
 class AutoParser(ColumnParser):
-    def __init__(self, partition_grammars: Optional[list[PartitionGrammar]] = None):
+    def __init__(self, partition_grammars: Optional[list[PartitionGrammar]] = None, parse_filenames: bool = False):
         self.partitions = partition_grammars
+        self._parse_filenames = parse_filenames
+        self._fname_column = "fname"  # default name, can be changed by setter
 
     def __call__(self, dirname: str) -> tuple[str, str]:
+        if self.parses_filenames() and "=" not in dirname:
```

Comment:
hm, this is a tricky case. At one side, I kinda like it -- it works in all the cases where you want it to work. At the other side, it exposes the flaw in the interface -- we _know_ at the callsite whether we process a file or a "directory", so we could just extend it, and not rely on this "hack".

I can imagine some cases where the mixing is seemingly valid -- say, your directory structure looks like this:
```
table1/year=../month=../...
table2/year=../month=../...
```
and the table1 and table2 have some columns in common thus it is legit to process their data together. *However*, even in this case I'd think it is a better idea to for-cycle through the tables, and launch a single query for each.

so I'd now lean slightly towards keeping this "hack" solution, rather than extending the interface -- until I see a valid example for the `=` and non-`=` mixing, I'd rather misbehave on that, to discourage users from what I consider poor design.